### ISSUE TITLE
Question for reproducing SELD based on  external dataset

### ISSUE THREAD

**Issue by zyc-c** [sharathadavanne/seld-dcase2022#3]
Dear professor,
      Hi. Sorry to bother you. Actually I am reproducing the baseline CRNN-Multi-accodoa network in DACSE 2022 for FOA dataset. However, I can not obtain the similar result compared with the result in your github. There is a enormous gap for the two results. Later, I found the reply for similar question,but I still do not understand the workflow in details.  Should I mix the synthesis split and real split directly into the same document  and then operating feature extraction? Or, I should first train the network for synthesis split and save the best model as pretrained model in 'parameter.py' and then test the real split based on the pretrained model? 
     Could you offer the detailed operation steps in the code when you are free if this is allowed? And I want to know that if the external dataset is necessary to use if i submit the results in DCASE2022?

**Comment by sharathadavanne** [comment]
Hi @zyc-c the baseline method uses the [following folds for training.](https://github.com/sharathadavanne/seld-dcase2022/blob/0bc1eaf2c0902f690b7285ca057976156176d407/train_seldnet.py#L235). Where, 
Fold1: synth-train (synthetic mixtures)
Fold2: synth-test (synthetic mixtures)
Fold3: devset-train (real-life recordings)
Fold4: devset-test (real-life recordings)
Fold5: evalset (real-life recordings)

Fold 1 & 2 are from https://zenodo.org/record/6406873#.YpjhChNBw-R 
Fold 3, 4 & 5 are from https://zenodo.org/record/6600531#.YpjhQRNBw-Q 

To obtain the baseline results please download Folds 1 to 4 and use the baseline code in this repo.

I hope this information helps.

**Comment by zyc-c** [comment]
Thanks! It is helpful.
However, I have a new problem when I run the train_seldnet.py using evaluation dataset.

**Comment by zyc-c** [comment]
Thanks! It is helpful.
However, I have a new problem when I run the train_seldnet.py using evaluation dataset.

**Comment by zyc-c** [comment]
Thanks! It is helpful.
However, I have a new problem when I run the train_seldnet.py using evaluation dataset.
When I run the train_seldnet.py,It provide a error : "TypeError:'NoneType' object is not iterable " The error appear at line 243 "for split_cnt ,split in enumerate(test_splits)".
I do not understand why the error will happen. After all, there is not null value existing. Before run it , I just change the mode from dev to eval in the whole code which is same operation when I run  the code offered DCASE2021. So I want to know if you have met it or have method solving it.

**Comment by sharathadavanne** [comment]
Hello @zyc-c as you [see here ](https://github.com/sharathadavanne/seld-dcase2022/blob/0bc1eaf2c0902f690b7285ca057976156176d407/train_seldnet.py#L224)in the baseline code we have only configured the test_splits for the 'dev' data. You will have to configure them for 'eval' data yourself, based on your training procedure.

**Comment by zyc-c** [comment]
Thank you very much! I have configured them for 'eval' data. But the results do not appear. The whole code do not show red error.&nbsp; But it tell me "Load best model weights -Loading unseen test dataset: Computing some stats about the dataset --Loading features failed" Maybe the eval_norm feature do not be read? I do don't understand it.&nbsp;
Further, I try change is_eval = False to True in cls_feature_class.py and cls_data_genetator.py, the new error appeared:"TypeError:'NoneType' object is not subscriptable " and it point to ''torch.nn.Linear(params['fnn_size'] if params['nb_fnn_layers'] else params['rnn_size'], out_shape[-1], bias=True)'' in line 251,seldnet_model.py . That is strange.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "sharathadavanne/seld-dcase2022"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月14日(星期二) 晚上7:44
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [sharathadavanne/seld-dcase2022] Question for reproducing SELD based on  external dataset (Issue #3)





 
Hello @zyc-c as you see here in the baseline code we have only configured the test_splits for the 'dev' data. You will have to configure them for 'eval' data yourself, based on your training procedure.

**Comment by zyc-c** [comment]
Sorry to bother you again.
refer to the code in DCASE2021. if I should set&nbsp; ''is_eval = true'' to ''data_gen_test = cls_data_generator.DataGenerator(
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; params=params, split=test_splits[split_cnt], shuffle=False, per_file=True
&nbsp; &nbsp; &nbsp; &nbsp; )'' The feature could seem be read, but&nbsp;


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "sharathadavanne/seld-dcase2022"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月14日(星期二) 晚上7:44
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [sharathadavanne/seld-dcase2022] Question for reproducing SELD based on  external dataset (Issue #3)





 
Hello @zyc-c as you see here in the baseline code we have only configured the test_splits for the 'dev' data. You will have to configure them for 'eval' data yourself, based on your training procedure.

**Comment by zyc-c** [comment]
Sorry to bother you again.
refer to the code in DCASE2021. if I should set&nbsp; ''is_eval = true'' to ''data_gen_test = cls_data_generator.DataGenerator(
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; params=params, split=test_splits[split_cnt], shuffle=False, per_file=True
&nbsp; &nbsp; &nbsp; &nbsp; )'' The feature could seem be read, but
a new error ''too many values to unpack(except 2)'' appears and no error localization.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "sharathadavanne/seld-dcase2022"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月14日(星期二) 晚上7:44
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [sharathadavanne/seld-dcase2022] Question for reproducing SELD based on  external dataset (Issue #3)





 
Hello @zyc-c as you see here in the baseline code we have only configured the test_splits for the 'dev' data. You will have to configure them for 'eval' data yourself, based on your training procedure.
